docs/guide/content/development_guide/architecture.md

```diff
@@ -0,0 +1,58 @@
+# Architecture and Code Structure
+
+OnDemand Loop is a Ruby on Rails application that runs as an Open OnDemand
+passenger app.  The project follows standard Rails conventions as long as they do
+not interfere with simplicity or functionality.  Instead of a traditional
+database it stores project and file metadata in YAML files on disk.  A small set
+of models like `Project`, `DownloadFile`, and `UploadBundle` inherit from
+`ApplicationDiskRecord` which provides YAML based persistence.
+
+Transfers are processed by background services started by
+`DetachedProcessManager`.  Each service loops over pending download or upload
+requests and spawns connector specific processors.  Status updates are recorded
+back to the YAML metadata so the web UI can poll and display progress.
+
+The web interface communicates with these services via simple command endpoints
+exposed through the `CommandRegistry`.  This design keeps the runtime
+lightweight while avoiding the need for a separate job queue.
+
+Connectors encapsulate repository specific logic and are loaded dynamically
+through `ConnectorClassDispatcher`. They make Loop extensible by providing
+custom processors and controllers for each repository. The reference
+implementation is the Dataverse connector, with Zenodo as an additional
+example. See the [Connectors](connectors.md) section for the exact features a
+connector must implement.
+
+## Project Layout
+
```

Comment:
I would also explain that the Rails version is 7.2.2.1 because Is the highest version supported by the Ruby version 3.1 integrated in this version of OOD